### PR TITLE
[Docs] Update Windows build instructions for using WSL and Ubuntu 17.04

### PR DESCRIPTION
**Pull request by fanquake**
This updates the Windows build documentation with the workaround required to build using Ubuntu 17.04 on WSL, and makes it's explicit that building on Ubuntu 16.04 is broken, and not recommended.

This includes a commit from @donaloconnor in #11244, and is mostly the investigative work of @laanwj throughout #8732, #8653 and quite a few other issues.

I tested building on 14.04, 16.04.3 and 17.04 [here](https://github.com/bitcoin/bitcoin/pull/11244#issuecomment-327990251) and got the results we expect.

---

Built master at https://github.com/bitcoin/bitcoin/commit/c22a53cd6351b3f0e96de971b001ad712b83fc47 on a Windows 10 VM (Version 1607, OS Build 14393.1593) using WSL with Ubuntu 14.04.
windows

Upgraded WSL to 16.04.3, and tried building https://github.com/bitcoin/bitcoin/commit/c22a53cd6351b3f0e96de971b001ad712b83fc47 using these instructions. The result is as expected.
ubuntu 16 04 3

Upgraded WSL to 17.04 and tried building https://github.com/bitcoin/bitcoin/commit/3255d6347b1f9eccbec3d6d93d4a424087a3b35b using these instructions.
ubuntu 17 04

If someone else could also verify that builds are working on both 14.04 and 17.04 with these instructions, that would be great.

### CODE REVIEW

File: doc/build-windows.md [hunk 1]
```diff
@@ -64,6 +65,13 @@ build process.
 
 See also: [dependencies.md](dependencies.md).
 
+If your building on Ubuntu 17.04, run these two commands, selecting the 'posix' variant for both,
```
Comment (MeshCollider) [inline doc/build-windows.md]: `your` -> `you're`

**Review by meshcollider — COMMENTED** [review]
Concept ACK, looks good. Will try and test soon.

**Comment by laanwj** [comment]
Thanks for doing this!
Hope someone that builds on windows can check this.

File: doc/build-windows.md [hunk 2]
```diff
@@ -4,10 +4,11 @@ WINDOWS BUILD NOTES
 Below are some notes on how to build Bitcoin Core for Windows.
 
 Most developers use cross-compilation from Ubuntu to build executables for
-Windows. This is also used to build the release binaries.
+Windows. Cross-compilation is also used to build the release binaries.
 
-Currently only building on Ubuntu Trusty 14.04 is supported.
-Other versions are unsupported or known to be broken (e.g. Ubuntu Xenial 16.04).
+Currently only building on Ubuntu Trusty 14.04 or Ubuntu Zesty 17.04 is supported.
```
Comment (MarcoFalke) [inline doc/build-windows.md]: Should say "17.04 and later"

File: doc/build-windows.md [hunk 3]
```diff
@@ -64,6 +65,13 @@ build process.
 
 See also: [dependencies.md](dependencies.md).
 
+If you're building on Ubuntu 17.04, run these two commands, selecting the 'posix' variant for both,
```
Comment (MarcoFalke) [inline doc/build-windows.md]: "17.04 or later"

**Review by maflcko — COMMENTED** [review]
Thanks! Tested ACK efd88160e39e5d55905eb6efa0c0f08029ef7a23 on artful.

**Comment by laanwj** [comment]
utACK 696ce46

**Pull request merged**